doc/api/path.md

```diff
@@ -177,57 +247,73 @@ path.isAbsolute('bar\\baz')  // false
 path.isAbsolute('.')         // false
 ```
 
-*Note:* If the path string passed as parameter is a zero-length string, unlike
-        other path module functions, it will be used as-is and `false` will be
-        returned.
+A [`TypeError`][] is thrown if `path` is not a string.
 
 ## path.join([path1][, path2][, ...])
 
-Join all arguments together and normalize the resulting path.
+* `path1` {String} A path segment
+* `path2` {String} A path segment
+* `[, ...]` {String} Additional path segments
+
+The `path.join()` method join all given path segments together using the
+platform specific separator (`path.sep`) as a delimiter, then normalizes
+the resulting path.
 
-All arguments must be strings.  In v0.8, non-string arguments were
-silently ignored.  In v0.10 and up, an exception is thrown.
+Zero-length strings passed as segments are ignored. If the joined path string is
+a zero-length string then `'.'` will be returned, representing the current
+working directory.
 
-Examples:
+For example:
 
 ```js
 path.join('/foo', 'bar', 'baz/asdf', 'quux', '..')
 // returns '/foo/bar/baz/asdf'
 
 path.join('foo', {}, 'bar')
-// throws exception
-TypeError: Arguments to path.join must be strings
+// throws TypeError: Arguments to path.join must be strings
 ```
 
-*Note:* If the arguments to `join` have zero-length strings, unlike other path
-        module functions, they will be ignored. If the joined path string is a
-        zero-length string then `'.'` will be returned, which represents the
-        current working directory.
+A [`TypeError`] is thrown if any of the path segments is not a string.
 
 ## path.normalize(path)
 
-Normalize a path, taking care of `'..'` and `'.'` parts. `path` must be a
-string.
+* `path` {String} A file path
+
+The `path.normalize()` method normalizes a path, resolving `'..'` and `'.'`
+segments.
+
+When multiple, sequential `/` (slash) characters are found, they are replaced
```

Comment:
This might be reworded to not mention specific slash characters and instead use wording like 'platform specific path segment separator' or something shorter (but still differentiates from the platform specific path separator -- `;` or `:`).